1-Draft/RFCNNNN-RFC-Native-Command-Exit-Errors.md

```diff
@@ -0,0 +1,236 @@
+---
+RFC: RFCNNNN-RFC-Native-Command-Exit-Errors
+Author: Micheal Padden
+Status: Draft
+Area: Engine, Language
+Version: 1.0.0
+---
+
+# Native Command Error Handling
+
+Although Powershell has an exception-based error handling framework, it
+does not currently apply to native commands. 
+
+Exception-based error handling makes it easier to write robust code
+as less boilerplate code is needed to check and handle errors.
+
+Powershell scripts using native commands would benefit from being able
+to error handling features like those used by cmdlets.
+
+## Motivation
+
+Native commands return an exit code to the calling application which
+will be zero for success or non-zero for failure. A robust script will
+not assume success, instead checking the exit code after any statement
+calling a native command, using boilerplate like the following:
+```Powershell
+ if( ! $? ) 
+ { 
+     exit $lastexitcode; 
+ }
+```
+
+Bourne type shells provide a `set -e` error handling mode that acts as 
+if this boilerplate were included in certain contexts. Like exception
+based error handling, this can exit a stack of functions, scripts and
+shells. This allows robust scripts to be written with a minimal amount
+of boilerplate.
+
+To support a similar style of error handling with native commands, this
+RFC proposes including native command errors in Powershell's exception
+handling framework in a similar way.
+
+The specification and alternative proposals are based on the
+[Equivalent of bash `set -e` #3415](https://github.com/PowerShell/PowerShell/issues/3415)
+and committee review of the associated
+[pull request](https://github.com/PowerShell/PowerShell/pull/3523).
+
+
+## Specification
+
+This RFC proposes including native commands in the error handling
+framework, by allowing an error to be reported to the error stream
+when a native command exits with a non-zero exit code, similar to 
+the `set -e` option in bourne type shells.
+
+The `$PSIncludeNativeCommandInErrorActionPreference` preference
```

Comment:
That's a hella long variable name

Reply:
Maybe `$NativeErrorInclusionPreference`? or `$NativeErrorPreference`? Since it's effectively an enum, the value could elucidate the effect: `$NativeErrorPreference = 'IncludeInError'` or something

Reply:
Maybe `$NonZeroExitCodeActionPreference`? It's a bit more self-explanatory.